contracts/AgentMech.sol

```diff
@@ -68,17 +78,28 @@ contract AgentMech is ERC721Mech {
         price = _price;
     }
 
-    /// @dev Registers a request.
-    /// @param data Self-descriptive opaque data-blob.
-    function request(bytes memory data) external payable returns (uint256 requestId) {
-        if (msg.value < price) {
+    /// @dev Performs actions before the request is posted.
+    /// @param amount Amount of payment in wei.
+    function _preRequest(uint256 amount, uint256, bytes memory) internal virtual {
+        // Check the request payment
+        if (amount < price) {
             revert NotEnoughPaid(msg.value, price);
         }
+    }
```

Comment:
Virtual pre-request function that can be changed by other contracts (for the default mech it just checks for the payment).